sdk/python/test_data/pipelines/pipeline_with_parallelfor_list_artifacts.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2022 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List
+
+from kfp import compiler
+from kfp import dsl
+from kfp.dsl import Artifact
+from kfp.dsl import Dataset
+
+
+@dsl.component
+def print_artifact_name(artifact: Artifact) -> str:
+    print(artifact.name)
+    return artifact.name
+
+
+@dsl.component
+def make_dataset(text: str) -> Dataset:
+    dataset = Dataset(uri=dsl.get_uri(), metadata={'length': len(text)})
+    with open(dataset.path, 'w') as f:
+        f.write(text)
+    return dataset
+
+
+@dsl.pipeline
+def make_datasets(
+        texts: List[str] = ['Hello', ',', ' ', 'world!']) -> List[Dataset]:
+    with dsl.ParallelFor(texts) as text:
+        t1 = make_dataset(text=text)
+
+    return dsl.Collected(t1.output)
+
+
+@dsl.component
+def make_artifact(name: str) -> Artifact:
```

Comment:
Since this is used as the artifact `content`, perhaps rename this parameter to something other than `name`. The artifact name is a different attribute on the artifact. If you look at the value printed by `print(artifact.name)` in downstream tasks you wont see the value you pass in here.

Reply:
Done, renamed to `data`.